src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonArchitecture3rdPartyDatatypeMappingsCheck.java

```diff
@@ -0,0 +1,34 @@
+package com.devonfw.ide.sonarqube.common.impl.check;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+import com.devonfw.ide.sonarqube.common.api.JavaType;
+
+/**
+ * {@link DevonArchitecture3rdPartyCheck} verifying that datatypes are properly mapped
+ * @author lniazman
+ *
+ */
+@Rule(key = "Devon4j:E6", name = "Devon 3rd Party Datatype Mappings Check", //
+	description = "Verify that datatypes are mapped to the JPA using the correct annotation", //
+	priority = Priority.CRITICAL, tags = { "architecture-violation" }) //
+public class DevonArchitecture3rdPartyDatatypeMappingsCheck extends DevonArchitecture3rdPartyCheck {
+
+  @Override
+  protected String checkDependency(JavaType source, JavaType target) {
+
+	if (target.toString().equals("org.hibernate.Type")) {
+      return "Use the javax.persistence.Converter annotation on a custom converter "
```

Comment:
your indentation is broken. The standard in devonfw is to have a properly configured formatter (e.g. indendation with 2 spaces) with save-actions in Eclipse. Are you using devon-ide with Eclipse?
Of course just a nice to have, but as this will impact all your contributions and PRs, please configure your IDE properly and assure code formatting is performed according to devonfw standards as otherwise we have diff wars in git and the diffs are hard to read for the reviewers.

Reply:
Yes, I'm not sure why this problem keeps popping up. My formatter profile is set to 'clientpe', which should be the correct profile with all the proper configurations. I'll look into it and see what's wrong. 
Thanks for the review and merge!